crates/findex/src/pyo3/mod.rs

```diff
@@ -10,14 +9,119 @@ macro_rules! pyo3_unwrap {
     };
 }
 
+/// Implements the basic functionalities of keyword and location in python.
+///
+/// # Parameters
+///
+/// - `type_name`   : name of the key type
+macro_rules! impl_python_byte {
+    ($py_type:ty, $rust_type:ty, $name:tt) => {
+        #[pymethods]
+        impl $py_type {
+            /// Create from bytes.
+            #[staticmethod]
+            pub fn from_bytes(val: &[u8]) -> Self {
+                Self(<$rust_type>::from(val))
+            }
+
+            /// Create from string.
+            #[staticmethod]
+            pub fn from_string(val: &str) -> Self {
+                Self(<$rust_type>::from(val))
+            }
+
+            /// Create from int.
+            #[staticmethod]
+            pub fn from_int(val: i64) -> Self {
+                Self(<$rust_type>::from(val.to_be_bytes().to_vec()))
+            }
+
+            /// Converts to string.
+            fn __str__(&self) -> PyResult<String> {
+                match String::from_utf8(self.0.to_vec()) {
+                    Ok(s) => Ok(s),
+                    Err(e) => Err(pyo3::exceptions::PyValueError::new_err(e.to_string())),
+                }
+            }
+
+            /// Converts to bytes.
+            fn __bytes__(&self, py: Python) -> Py<PyBytes> {
+                PyBytes::new(py, &self.0).into()
+            }
+
+            /// Converts to int.
+            fn __int__(&self) -> PyResult<i64> {
```

Comment:
Why do you need to be able to convert into a i64?

I would rather let the user transform bytes into the specific type in use.

Reply:
to match the functionalities of JS and Java libs.